slave_stack/examples/nuttx_arduino_due/main.cc

```diff
@@ -0,0 +1,75 @@
+
+#include "nuttx/SPI.h"
+#include "kickcat/ESC/Lan9252.h"
+
+using namespace kickcat;
+
+void reportError(hresult const& rc)
+{
+    if (rc != hresult::OK)
+    {
+        printf("%s\n", toString(rc));
+    }
+}
+
+void esc_routine(Lan9252& esc)
+{
+    uint32_t nb_bytes = 32;
+    uint8_t test_write[nb_bytes];
+    for (uint32_t i=0; i < nb_bytes; ++i)
+    {
+        test_write[i] = i;
+    }
+    reportError(esc.write(0x1200, &test_write, nb_bytes));
+
+    uint16_t al_status;
+    reportError(esc.read(AL_STATUS, &al_status, sizeof(al_status)));
+    bool watchdog = false;
+    reportError(esc.read(WDOG_STATUS, &watchdog, 1));
+
+//    if ((al_status & ESM_OP) and watchdog)
+//    {
+//        uint8_t test_read[nb_bytes];
+//        reportError(esc.read(0x1000, &test_read, nb_bytes));
+//        for (uint32_t i=0; i < nb_bytes; i++)
+//        {
+//            printf("%x ", test_read[i]);
+//        }
+//        printf(" received\n");
+//    }
+}
+
+extern "C" int main(int argc, char *argv[])
```

Comment:
I do not think you need extern "C" with @leducp we tested and cpp int main works :)